src/OgDeleteOrphansBase.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Drupal\og;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Base implementation for OgDeleteOrphans plugins.
+ */
+abstract class OgDeleteOrphansBase implements OgDeleteOrphansInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function register(EntityInterface $entity) {
+    throw new \Exception(__METHOD__ . ' is not implemented.');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function query() {
+    throw new \Exception(__METHOD__ . ' is not implemented.');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function configurationForm($form, FormStateInterface $form_state) {
```

Comment:
I think it might be too much (i.e. give site builers a way to shot their foot). Lets go with some sane number (50), and of someone wants to have control over this - let them extend the plugin?